[WIP] Scripts/Northrend: Move Sholazar Basin NPCs to SmartAI

mysterious-static opened the pull request:
**Changes proposed:**

- move Avatar of Freya to SAI
- move Adventurous Dwarf to SAI
- move Shaman Vekjik to SAI
- move Bushwhacker to SAI
- update outdated list of scripts in, and remove unneeded includes from, zone_sholazar_basin.cpp

**Target branch(es):**

- [X] 3.3.5
- [X] master

**Tests performed:** (Does it build, tested in-game, etc.)
- builds
- sql is syntactically sound.

**Known issues and TODO list:** (add/remove lines as needed)

- [ ] Bushwhacker is not complete

Commit:
set up todos and data I need to write sai

Commit:
shaman vekjik

Commit:
conditions for gossip menu text

Commit:
Freya migrate and start header cleanup

Commit:
more cleanup

Commit:
remove adventurous dwarf

Commit:
remove include

Commit:
quote

tkrokli commented:
Just a minor detail: the last 3 blank lines in your SQL file, are they some sort of "this space intentionally left blank", meaning that you intent to add more content there?  :-)

Commit:
Update 9999_99_99_99_world_335.sql

velinath commented:
nope, just an error, I had completed this script to the best of my knowledge

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,34 @@
+(27801, 0, 0, 0, 62, 0, 100, 0, 9722, 0, 0, 0, 85, 52045, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Avatar of Freya - On Gossip Option 0 Selected - Invoker Cast \'Freya Conversation Credit\''),
+(28604, 0, 0, 3, 62, 0, 100, 0, 9724, 0, 0, 0, 11, 52073, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option 0 Selected - Cast \'Dwarf Gives Orange\''),
+(28604, 0, 1, 3, 62, 0, 100, 0, 9724, 1, 0, 0, 11, 52074, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option 1 Selected - Cast \'Dwarf Gives Banana\''),
+(28604, 0, 2, 3, 62, 0, 100, 0, 9724, 2, 0, 0, 11, 52076, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option 2 Selected - Cast \'Dwarf Gives Papaya\''),
```
Missing target.

tkrokli replied:
Good catch, thanks for checking. 👍

velinath replied:
This one's fixed. Thank you!

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,34 @@
+(28604, 0, 1, 3, 62, 0, 100, 0, 9724, 1, 0, 0, 11, 52074, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option 1 Selected - Cast \'Dwarf Gives Banana\''),
+(28604, 0, 2, 3, 62, 0, 100, 0, 9724, 2, 0, 0, 11, 52076, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option 2 Selected - Cast \'Dwarf Gives Papaya\''),
+(28604, 0, 3, 4, 61, 0, 100, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option 0 Selected - Say Line 1'),
+(28604, 0, 4, 0, 61, 0, 100, 0, 0, 0, 0, 0, 41, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option 0 Selected - Despawn Instant'),
```
According to this video https://youtu.be/07qr5SlBDk0?t=36s the npc doesn't despawn instant, you must close the gossip on gossip select, remove the npc flag, say the text, move forward 8 yards and despawn after 4 secs.

velinath replied:
I'll take care of this one tomorrow morning first thing.

Commit:
Update 9999_99_99_99_world_335.sql

Commit:
redo script based on video from Killyana

I will test this tonight

Commit:
Update 9999_99_99_99_world_335.sql

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,37 @@
+(28604, 0, 3, 4, 61, 0, 100, 0, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Close Gossip'),
+(28604, 0, 4, 5, 61, 0, 100, 0, 0, 0, 0, 0, 83, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Remove Gossip Flag'),
+(28604, 0, 5, 6, 61, 0, 100, 0, 0, 0, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Say Line 1'),
+(28604, 0, 6, 7, 61, 0, 100, 0, 0, 0, 0, 0, 46, 8, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Move Forward 8 Yards'),
```
This action was changed 46 is no longer used, you must use SMART_ACTION_MOVE_OFFSET:114

velinath replied:
sorry about this! i am still using the old public version of keira to do a lot of this and the action list is a bit out of date sometimes. are parameters the same? [wiki](https://trinitycore.atlassian.net/wiki/spaces/tc/pages/2130108/smart+scripts) doesn't specify for 114

velinath replied:
er nevermind I see it says x, y, z for offset, would that be just 8 yds on X axis in current orientation? sorry, it's just not clear to me intuitively

Killyana replied:
Yes it could be use to rise the npc in the air or move in any direction.

Killyana commented:
I made some changes to get the exactly the same retail behavior:
``` sql
-- Vekjik, Avatar of Freya, Adventurous Dwarf
UPDATE `creature_template` SET `AIName` = 'SmartAI', `ScriptName` = '' WHERE `entry` IN (27801, 28315, 28604);

DELETE FROM `gossip_menu_option` WHERE `MenuID` = 9724 AND `OptionID` IN (0, 2); -- 1 is already present in DB
INSERT INTO `gossip_menu_option` (`MenuID`, `OptionID`, `OptionIcon`, `OptionText`, `OptionBroadcastTextID`, `OptionType`, `OptionNpcFlag`, `ActionMenuID`, `ActionPoiID`, `BoxCoded`, `BoxMoney`, `BoxText`, `BoxBroadcastTextID`, `VerifiedBuild`) VALUES
(9724, 0, 0, 'Can you spare an orange?', 28594, 1, 1, 0, 0, 0, 0, '', 0, 0),
(9724, 2, 0, 'I could really use a papaya.', 28596, 1, 1, 0, 0, 0, 0, '', 0, 0);

UPDATE `creature_text` SET `emote`=5 WHERE `CreatureID`=28604 AND `GroupID`=0 AND `ID`=0;
UPDATE `creature_text` SET `emote`=3 WHERE `CreatureID`=28604 AND `GroupID`=1 AND `ID`=0;

DELETE FROM `smart_scripts` WHERE `source_type` = 0 AND `entryorguid` IN (27801, 28315, 28604);
DELETE FROM `smart_scripts` WHERE `source_type` = 9 AND `entryorguid` IN (2860400);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(28315, 0, 0, 1, 62, 0, 100, 0, 9686, 0, 0, 0, 72, 9686, 13138, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Shaman Vekjik - On Gossip Option 0 Selected - Close Gossip'),
(28315, 0, 1, 2, 61, 0, 100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Shaman Vekjik - On Gossip Option 0 Selected - Say Line 0'),
(28315, 0, 2, 3, 61, 0, 100, 0, 0, 0, 0, 0, 15, 12573, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Shaman Vekjik - On Gossip Option 0 Selected - Quest Credit \'Making Peace\''),
(28315, 0, 3, 0, 61, 0, 100, 0, 0, 0, 0, 0, 11, 51469, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Shaman Vekjik - On Gossip Option 0 Selected - Cast \'Frenzyheart\'s Fury\''),
(27801, 0, 0, 0, 62, 0, 100, 0, 9722, 0, 0, 0, 85, 52045, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Avatar of Freya - On Gossip Option 0 Selected - Invoker Cast \'Freya Conversation Credit\''),
(28604, 0, 0, 3, 62, 0, 100, 0, 9724, 0, 0, 0, 85, 52073, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option 0 Selected - Cast \'Dwarf Gives Orange\''),
(28604, 0, 1, 3, 62, 0, 100, 0, 9724, 1, 0, 0, 85, 52074, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option 1 Selected - Cast \'Dwarf Gives Banana\''),
(28604, 0, 2, 3, 62, 0, 100, 0, 9724, 2, 0, 0, 85, 52076, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option 2 Selected - Cast \'Dwarf Gives Papaya\''),
(28604, 0, 3, 0, 61, 0, 100, 0, 0, 0, 0, 0, 80, 2860400, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Close Gossip'),
(2860400, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Close Gossip'),
(2860400, 9, 1, 0, 0, 0, 100, 0, 0, 0, 0, 0, 83, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Remove Gossip Flag'),
(2860400, 9, 2, 0, 0, 0, 100, 0, 0, 0, 0, 0, 66, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Set Orientation'),
(2860400, 9, 3, 0, 0, 0, 100, 0, 2000, 2000, 0, 0, 1, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Say Line 1'),
(2860400, 9, 4, 0, 0, 0, 100, 0, 2000, 2000, 0, 0, 114, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 25, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Move Forward 8 Yards'),
(2860400, 9, 5, 0, 0, 0, 100, 0, 3000, 3000, 0, 0, 41, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Despawn In 4000 ms'),
(28604, 0, 4, 0, 1, 0, 100, 1, 1000, 1000, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - Occ no repeat - Say Line 0');

DELETE FROM `conditions` WHERE `SourceTypeOrReferenceId` IN (14, 15) AND `SourceGroup` IN (9678, 9720, 9724);
INSERT INTO `conditions` (`SourceTypeOrReferenceId`, `SourceGroup`, `SourceEntry`, `SourceId`, `ElseGroup`, `ConditionTypeOrReference`, `ConditionTarget`, `ConditionValue1`, `ConditionValue2`, `ConditionValue3`, `NegativeCondition`, `ErrorType`, `ErrorTextId`, `ScriptName`, `Comment`) VALUES
(14, 9678, 13136, 0, 0, 9, 0, 12573, 0, 0, 1, 0, 0, '', 'Show gossip menu 9678 text id 13136 if quest Making Peace has not been taken.'),
(14, 9678, 13137, 0, 0, 9, 0, 12573, 0, 0, 0, 0, 0, '', 'Show gossip menu 9678 text id 13137 if quest Making Peace has been taken.'),
(15, 9678, 0, 0, 0, 9, 0, 12573, 0, 0, 0, 0, 0, '', 'Show gossip menu 9678 option id 0 if quest Making Peace has been taken.'),
(15, 9720, 0, 0, 0, 9, 0, 12621, 0, 0, 0, 0, 0, '', 'Show gossip menu 9720 option id 0 if quest Freya\'s Pact has been taken.'),
(15, 9724, 0, 0, 0, 9, 0, 12634, 0, 0, 0, 0, 0, '', 'Show gossip menu 9724 option id 0 if quest Some Make Lemonade, Some Make Liquor has been taken.'),
(15, 9724, 0, 0, 0, 2, 0, 38656, 1, 0, 1, 0, 0, '', 'Show gossip menu 9724 option id 0 if player does not have 1 of Orange.'),
(15, 9724, 1, 0, 0, 9, 0, 12634, 0, 0, 0, 0, 0, '', 'Show gossip menu 9724 option id 1 if quest Some Make Lemonade, Some Make Liquor has been taken.'),
(15, 9724, 1, 0, 0, 2, 0, 38653, 2, 0, 1, 0, 0, '', 'Show gossip menu 9724 option id 1 if player does not have 2 of Banana Bunch.'),
(15, 9724, 2, 0, 0, 9, 0, 12634, 0, 0, 0, 0, 0, '', 'Show gossip menu 9724 option id 2 if quest Some Make Lemonade, Some Make Liquor has been taken.'),
(15, 9724, 2, 0, 0, 2, 0, 38655, 1, 0, 1, 0, 0, '', 'Show gossip menu 9724 option id 2 if player does not have 1 of Papaya.');

velinath commented:
Oh nice, that looks awesome. Thank you for putting that together! I would not have thought about timed actionlists, but that's probably because I'm still operating at a pretty basic level w/r/t smart scripting. It makes a lot of sense now that I am actually looking at it. I'll update the SQL shortly.

On another note I was looking further into npc movement after your note earlier and I would also like to work on npc_bushwhacker as well before this gets merged as I believe I understand how to script his behavior at this time. I will look at his script this afternoon when I have access to a server I can test against.

Killyana commented:
Yes npc_bushwhacker could be moved to SAI.

Commit:
Updates to include actionlist

per @Killyana

Killyana commented:
I updates the script to fix  the comments.

Commit:
Update 9999_99_99_99_world_335.sql

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,43 @@
+(28315, 0, 1, 2, 61, 0, 100, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Shaman Vekjik - On Gossip Option 0 Selected - Say Line 0'),
+(28315, 0, 2, 3, 61, 0, 100, 0, 0, 0, 0, 0, 15, 12573, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Shaman Vekjik - On Gossip Option 0 Selected - Quest Credit \'Making Peace\''),
+(28315, 0, 3, 0, 61, 0, 100, 0, 0, 0, 0, 0, 11, 51469, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Shaman Vekjik - On Gossip Option 0 Selected - Cast \'Frenzyheart\'s Fury\''),
+(27801, 0, 0, 0, 62, 0, 100, 0, 9722, 0, 0, 0, 85, 52045, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Avatar of Freya - On Gossip Option 0 Selected - Invoker Cast \'Freya Conversation Credit\''),
```
Close the gossip here.

velinath replied:
also for 28315 I imagine?

Commit:
first pass on bushwhacker, gonna tweak later today

specifically i need to verify how and when the npc targets the summoner
/ whether it can aggro on other players when it spawns or whether it
will always target the summoner

Commit:
Update 9999_99_99_99_world_335.sql

Commit:
Merge branch 'sholazar-sai' of https://github.com/velinath/trinitycore into sholazar-sai

Commit:
Update 9999_99_99_99_world_335.sql

Commit:
bushwhacker

known issues: bushwhacker still spawns in the wrong spot and sometimes won't go up on the log. comment forthcoming.

velinath commented:
source: https://www.youtube.com/watch?v=Mkwt-kkyXqs

still missing: i need to add what looks like about a 2 second delay after he gets to the player for start attack
```sql
(28317, 0, 1, 0, 9, 0, 100, 0, 0, 3, 2000, 2000, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - Within 0-3 Yards - Start Attacking Summoner');
``` 
maybe?

the creature also spawns in the wrong area and won't go up the log as in the video. i'm not sure how to modify the creature to make it behave properly

Commit:
Fixes spawn point for Bushwhacker

velinath commented:
Thanks, that worked perfect!

Is the proposed fix for the creature behavior valid?

Killyana commented:
Nop, the npc must use immune to pc+npc on summon and once the point reached it removes the unit flags and attacks.

velinath commented:
so ```sql
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(28317, 0, 0, 1, 54, 0, 100, 0, 1, 1, 0, 0, 69, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Summon - Move To Summoner'),
(28317, 0, 1, 0, 61, 0, 100, 0, 0, 0, 0, 0, 18, 163842, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Summon - Set Flags Not Attackable & Pacified'),
(28317, 0, 2, 0, 34, 0, 100, 0, 0, 0, 0, 0, 80, 2831700, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Movement Inform - Run Timed Actionlist 2831700'),
(2831700, 9, 0, 0, 0, 0, 100, 0, 1500, 1500, 0, 0, 19, 131074, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - Timed Actionlist - Remove Flags Not Attackable & Pacified'),
(2831700, 9, 1, 0, 0, 0, 100, 0, 1500, 1500, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - Timed Actionlist - Start Attacking Summoner');
```

Killyana commented:
I did this, but there's probably a missing spell used in combat:
``` sql
UPDATE `creature_template` SET `unit_flags` = 768 WHERE `entry` = 28317;
DELETE FROM `smart_scripts` WHERE `source_type` = 0 AND `entryorguid` IN (28317);
DELETE FROM `smart_scripts` WHERE `source_type` = 9 AND `entryorguid` IN (2831700);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(28317, 0, 0, 0, 54, 0, 100, 0, 0, 0, 0, 0, 69, 1, 0, 0, 2, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Summon - Move To Summoner'),
(28317, 0, 1, 0, 34, 0, 100, 0, 0, 1, 0, 0, 80, 2831700, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Movement Inform - Run Timed Actionlist 2831700'),
(2831700, 9, 0, 0, 0, 0, 100, 0, 1500, 1500, 0, 0, 19, 768, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - Timed Actionlist - Remove Flags immune to pc/npc'),
(2831700, 9, 1, 0, 0, 0, 100, 0, 1500, 1500, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - Timed Actionlist - Start Attacking Summoner'),
(28317, 0, 2, 0, 7, 0, 100, 0, 0, 0, 0, 0, 41, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On evade - Despawn');

Commit:
Script for Bushwhacker

Thanks to @Killyana for the guidance on this. It's very appreciated.

velinath commented:
Went ahead and went with that for now. I used a bitwise operator in the commit for setting unit_flags since it already had 32768.

Commit:
Insert creature text.

Commit:
Bushwhacker play emote when reaches point

Killyana commented:
I update the script.

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,61 @@
+(28317, 0, 0, 0, 54, 0, 100, 0, 1, 1, 0, 0, 69, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Summon - Move To Summoner'),
+(28317, 0, 1, 2, 34, 0, 100, 0, 0, 0, 0, 0, 5, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Movement Inform - Play Emote 0'),
+(28317, 0, 2, 0, 61, 0, 100, 0, 0, 0, 0, 0, 80, 2831700, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Movement Inform - Run Timed Actionlist 2831700'),
+(2831700, 9, 0, 0, 0, 0, 100, 0, 1500, 1500, 0, 0, 19, 768, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - Timed Actionlist - Remove Flags Immune to PC/NPC'),
```
Move the text here.

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,61 @@
+(2860400, 9, 5, 0, 0, 0, 100, 0, 3000, 3000, 0, 0, 41, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Action list - Despawn In 3000 ms'),
+(28604, 0, 4, 0, 1, 0, 100, 1, 1000, 1000, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - OOC No Repeat - Say Line 0'),
+(28317, 0, 0, 0, 54, 0, 100, 0, 1, 1, 0, 0, 69, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Summon - Move To Summoner'),
+(28317, 0, 1, 2, 34, 0, 100, 0, 0, 0, 0, 0, 5, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Movement Inform - Play Emote 0'),
```
Add no repeat flag here.

velinath commented:
Will update when I get back to a computer. I originally used a link instead of putting the emote in the timed actionlist (see last commit) - for my future reference, is it TC convention to use the actionlist for all relevant actions if one is present?

Killyana commented:
If there's many actions that needs to be timed, in this case we use an action list.

Commit:
Update Bushwhacker script.

moves emote to actionlist, adjust timers based on the video

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -41,11 +41,11 @@ INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_
-(2831700, 9, 0, 0, 0, 0, 100, 0, 1500, 1500, 0, 0, 19, 768, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - Timed Actionlist - Remove Flags Immune to PC/NPC'),
-(2831700, 9, 1, 0, 0, 0, 100, 0, 1500, 1500, 0, 0, 49, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - Timed Actionlist - Start Attacking Summoner'),
-(28317, 0, 3, 0, 7, 0, 100, 0, 0, 0, 0, 0, 41, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On evade - Despawn');
+(28317, 0, 1, 0, 34, 0, 100, 1, 0, 0, 0, 0, 80, 2831700, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Movement Inform - Run Timed Actionlist 2831700'),
```
keep the emote and the 1.5secs timer for the action just put the text action at id=1

velinath replied:
I adjusted the timers since the flag being removed and attack start appear to happen at the same time in the video, and with both of the events in SAI having timers they were happening staggered.

To clarify emote id 0 also should be played as part of the actionlist?

Killyana replied:
Ok , just add the emote on just summoned if you want, the rest looks good.

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,61 @@
+(2860400, 9, 4, 0, 0, 0, 100, 0, 2000, 2000, 0, 0, 114, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 25, 0, 0, 'Adventurous Dwarf - On Action list - Move offset 25 Yards'),
+(2860400, 9, 5, 0, 0, 0, 100, 0, 3000, 3000, 0, 0, 41, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Action list - Despawn In 3000 ms'),
+(28604, 0, 4, 0, 1, 0, 100, 1, 1000, 1000, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - OOC No Repeat - Say Line 0'),
+(28317, 0, 0, 0, 54, 0, 100, 0, 1, 1, 0, 0, 69, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Summon - Move To Summoner'),
```
Just summoned don't use params

velinath replied:
i just noticed also that you had updated positioning on this event, I'll get that in commit upcoming shortly as well

Killyana replied:
Yes also contact distance =2

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,61 @@
+(28604, 0, 3, 0, 61, 0, 100, 0, 0, 0, 0, 0, 80, 2860400, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Gossip Option Selected - Action list'),
+(2860400, 9, 0, 0, 0, 0, 100, 0, 0, 0, 0, 0, 72, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Action list - Close Gossip'),
+(2860400, 9, 1, 0, 0, 0, 100, 0, 0, 0, 0, 0, 83, 1, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Action list - Remove Gossip Flag'),
+(2860400, 9, 2, 0, 0, 0, 100, 0, 0, 0, 0, 0, 66, 0, 0, 0, 0, 0, 0, 7, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Action list - Set Orientation'),
```
Add PointId=1 here

Killyana commented on sql/updates/world/3.3.5/9999_99_99_99_world_335.sql:
```diff
@@ -0,0 +1,61 @@
+(2860400, 9, 5, 0, 0, 0, 100, 0, 3000, 3000, 0, 0, 41, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - On Action list - Despawn In 3000 ms'),
+(28604, 0, 4, 0, 1, 0, 100, 1, 1000, 1000, 0, 0, 1, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Adventurous Dwarf - OOC No Repeat - Say Line 0'),
+(28317, 0, 0, 0, 54, 0, 100, 0, 1, 1, 0, 0, 69, 0, 0, 0, 0, 0, 0, 23, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Summon - Move To Summoner'),
+(28317, 0, 1, 0, 34, 0, 100, 1, 0, 0, 0, 0, 80, 2831700, 2, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 0, 0, 'Bushwhacker - On Movement Inform - Run Timed Actionlist 2831700'),
```
event_param1=1

Commit:
more touch-ups on Bushwhacker

velinath commented:
After looking at it some more I'm not convinced that emote action does anything at all, it could probably be taken out

Killyana committed:
Update 9999_99_99_99_world_335.sql

Killyana committed:
Update 9999_99_99_99_world_335.sql

Aokromes committed:
Rename 9999_99_99_99_world_335.sql to 2017_11_08_00_world.sql
